src/Test/FieldConstraintsSetFactoryTestCase.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AssoConnect\ValidatorBundle\Test;
+
+use AssoConnect\ValidatorBundle\Validator\ConstraintsSetFactory\Field\FieldConstraintsSetFactoryInterface;
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * This test should test validation violations instead of the returned set of constraints
+ * But that would require using the Symfony Compound constraint which cannot be used for tests
+ * @link https://github.com/symfony/symfony/issues/50205
+ */
+abstract class FieldConstraintsSetFactoryTestCase extends TestCase
+{
+    abstract protected function getFactory(): FieldConstraintsSetFactoryInterface;
+
+    /**
+     * @param mixed[] $fieldMapping
+     * @param Constraint[] $constraints
+     * @dataProvider getConstraintsForTypeProvider
+     */
+    public function testGetConstraintsForType(array $fieldMapping, array $constraints): void
+    {
+        $factory = $this->getFactory();
+        self::assertTrue($factory->supports($fieldMapping['type']));
+
+        self::assertArrayContainsSameObjects(
```

Comment:
With this test, I feel like we are testing the implementation of code.  Can it be a good test to validate values with constraints provided by the factory?

Reply:
@MPur Yes you're right but we can't right now cf the class comment ;) 
>  This test should test validation violations instead of the returned set of constraints
> But that would require using the Symfony Compound constraint which cannot be used for tests
> @link https://github.com/symfony/symfony/issues/50205